From: Simon McVittie <smcv@debian.org>
Date: Thu, 31 Mar 2016 14:32:55 +0100
Subject: Probe for GNU parallel more accurately

moreutils parallel isn't compatible with the command-line syntax
used here. Because it doesn't implement GNU-style --help and exits 1
when that option is given, this test was correctly skipped when
using moreutils parallel, but only by mistake.

moreutils parallel might conceivably gain --help and --version in
future, but hopefully nothing incompatible with GNU parallel is going
to gain a --gnu option.

Also use the --gnu option to force the new command-line semantics;
some versions optionally supported an incompatible command-line syntax
taken from moreutils parallel.

Signed-off-by: Simon McVittie <smcv@debian.org>
Applied-upstream: 2016.5, https://github.com/ostreedev/ostree/commit/9dafc820241ed6e663ad4093660fc255f549aa26
---
 tests/test-admin-locking.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/test-admin-locking.sh b/tests/test-admin-locking.sh
index 564295d..01814f9 100755
--- a/tests/test-admin-locking.sh
+++ b/tests/test-admin-locking.sh
@@ -25,8 +25,8 @@ set -euo pipefail
 setup_os_repository "archive-z2" "syslinux"
 
 # If parallel is not installed, skip the test
-if ! parallel --help >/dev/null 2>&1; then
-    echo "1..0 # SKIP no /usr/bin/parallel"
+if ! parallel --gnu /bin/true </dev/null >/dev/null 2>&1; then
+    echo "1..0 # SKIP GNU parallel not available"
     exit 0
 fi    
 
@@ -42,7 +42,7 @@ echo "rev=${rev}"
 ${CMD_PREFIX} ostree admin deploy --karg=root=LABEL=MOO --karg=quiet --os=testos testos:testos/buildmaster/x86_64-runtime
 assert_has_dir sysroot/boot/ostree/testos-${bootcsum}
 
-parallel_cmd=parallel
+parallel_cmd="parallel --gnu"
 if parallel --help | grep -q -e --no-notice; then
     parallel_cmd="${parallel_cmd} --no-notice"
 fi
